pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -142,6 +144,86 @@ def _save_input_params(self, idata) -> None:
         idata.attrs["validate_data"] = json.dumps(self.validate_data)
         idata.attrs["yearly_seasonality"] = json.dumps(self.yearly_seasonality)
 
+    def _create_likelihood_distribution(
+        self,
+        dist: Dict,
+        mu: SharedVariable,
+        observed: Union[np.ndarray, pd.Series],
+        dims: str,
+    ) -> TensorVariable:
+        """
+        Create and return a likelihood distribution for the model.
+
+        This method prepares the distribution and its parameters as specified in the
+        configuration dictionary, validates them, and constructs the likelihood
+        distribution using PyMC.
+
+        Parameters
+        ----------
+        dist : Dict
+            A configuration dictionary that must contain a 'dist' key with the name of
+            the distribution and a 'kwargs' key with parameters for the distribution.
+        observed : Union[np.ndarray, pd.Series]
+            The observed data to which the likelihood distribution will be fitted.
+        dims : str
+            The dimensions of the data.
+
+        Returns
+        -------
+        TensorVariable
+            The likelihood distribution constructed with PyMC.
+
+        Raises
+        ------
+        ValueError
+            If 'kwargs' key is missing in `dist`, or the parameter configuration does
+            not contain 'dist' and 'kwargs' keys, or if 'mu' is present in the nested
+            'kwargs'
+        """
+        # Validate that 'kwargs' is present and is a dictionary
+        if "kwargs" not in dist or not isinstance(dist["kwargs"], dict):
+            raise ValueError(
+                "The 'kwargs' key must be present in the 'dist' dictionary and be a dictionary itself."
+            )
+
+        if "mu" in dist["kwargs"]:
+            raise ValueError(
+                "The 'mu' key is not allowed directly within 'kwargs' of the main distribution as it is reserved."
+            )
```

Comment:
This assumes that dist has a `mu` kwarg. If we have that restriction we should pre-select the types of distributions that are actually allowed and make sure `dist["dist"]` is one of those.

Reply:
For the likelihood distribution, we have by default `mu=mu_var`, by consequence I need to evaluate if `mu` is there. They can have any kind of distribution but `mu` will be a reserved value no matter what.

But if you define your likelihood with other distributions, you are allowed to use `mu` on them if you need.

Reply:
How does it work if my likelihood doesn't have a `mu` parameter? How does the MMM pass the mean to this distribution then?

Reply:
Copy!

Reply:
I created a restriction so that only distributions that use "mu" could be passed, based on our code, the ones I added for now are **Normal, StudentT, Laplace, Logistic, Wald, TruncatedNormal, or VonMises**.

I think they are the most typical too, but we can expand. Maybe @juanitorduz or @nialloulton have others in mind, in that case, we can add them.

Reply:
i cant see a scenario where the likelihood dep var won't be specified by mu. The only thing I've catered for in my personal cowboy branch is also the lognormal and hurdlelognormal. which are also specified by mu, but i have pt.log(mu) for these (as this ensures the model is not then multiplicative) . These are particularly useful likelihoods

Reply:
i have this for those scenarios, it's a little messy in my implementation, but they are useful likelihoods. Ultimately they are still specified by mu though :

 ```
   def create_likelihood(self, model_config, target_, mu):
        likelihood_config = model_config.get("likelihood", {})
        likelihood_type = likelihood_config.get("type")
        dims = likelihood_config.get("dims")

        if not likelihood_type:
            raise ValueError("Likelihood type must be specified in the model config.")

        likelihood_func = getattr(pm, likelihood_type, None)
        if likelihood_func is None:
            raise ValueError(f"Invalid likelihood type {likelihood_type}")

        # Transform mu if the likelihood type is Lognormal or HurdleLognormal
        if likelihood_type in ['LogNormal', 'HurdleLogNormal']:
            mu = pt.log(mu)

        # Create sub-priors
        sub_priors = {}
        for param, config in likelihood_config.items():
            if param not in ['type', 'dims']:  # Skip 'type' and 'dims'
                if param == 'params':  # Handle nested 'params'
                    for sub_param, sub_config in config.items():
                        sub_priors[sub_param] = self.create_priors_from_config({sub_param: sub_config})[sub_param]
                else:
                    sub_priors[param] = self.create_priors_from_config({param: config})[param]

        return likelihood_func(name="likelihood", mu=mu, observed=target_, dims=dims,  **sub_priors)
```

Reply:
Might consider Gamma as well, but not convinced about the different linking that Niall is playing with. I suggest not trying too many things for now and see what people actually pick up?

Not a strong opinion either way

Reply:
yeah i agree keep it relatively simple. If people are straying out into strange likelihoods, they're probably building custom pymc models anyway

Reply:
Asymmetric Laplace is the only other one i can think of, which allows for quantile regression? 

https://bambinos.github.io/bambi/notebooks/quantile_regression.html

Reply:
Ok Then, I'll add tomorrow Asymmetric Laplace and Gamma for now.